### DISCUSSION TITLE
What is the purpose of the `configLookup` option in the `translateOptions` function?

### DISCUSSION THREAD

**Discussion by amareshsm** [eslint/eslint#18904] [General] [answered]
In the `translateOptions` function, there's an option called [`configLookup`](https://github.com/eslint/eslint/blob/71f37c5bf04afb704232d312cc6c72c957d1c14e/lib/cli.js#L111C5-L111C17) within the first argument object. However, I am unable to determine where the value for `configLookup` is being set. While I can see it being used inside the `translateOptions` function, I haven’t been able to find any reference or source that assigns a value to `configLookup`.

Usage of `configLookup` option:
```js
if (configType === "flat") {
    overrideConfigFile = (typeof config === "string") ? config : !configLookup;
    if (overrideConfigFile === false) {
        overrideConfigFile = void 0;
    }
}
```
 If config is not a string, it checks whether configLookup is falsy (e.g., not found or false). If configLookup is falsy, it sets overrideConfigFile to false.
If overrideConfigFile is false: The subsequent if statement resets overrideConfigFile to undefined (void 0), effectively disabling the configuration file lookup.

**Comment by mdjermanovic** [comment, marked as answer]
> where the value for `configLookup` is being set

It's set by `optionator`:

https://github.com/eslint/eslint/blob/71f37c5bf04afb704232d312cc6c72c957d1c14e/lib/options.js#L83-L88

By default is `true`. If CLI flag `--no-config-lookup` was passed, it will be `false`.
